src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -636,6 +645,39 @@ DbResultSet dbResultSet
             return resultProperties;
         }
 
+        /// <summary>
+        /// Reads data into jsonString.
+        /// </summary>
+        /// <param name="dbDataReader">DbDataReader.</param>
+        /// <param name="availableSize">Available buffer.</param>
+        /// <param name="jsonString">jsonString to read into.</param>
+        /// <returns>size of data read in bytes.</returns>
+        public int StreamData(DbDataReader dbDataReader, int availableSize, StringBuilder jsonString)
+        {
+            int fieldSize = (int)dbDataReader.GetChars(0, 0, null, 0, 0);
+
+            // if the size of the field is less than available size, then we can read the entire field.
+            // else we throw exception.
+            ValidateSize(availableSize, fieldSize);
+
+            char[] buffer = new char[fieldSize];
+
+            // read entire field into buffer and reduce available size.
+            dbDataReader.GetChars(0, 0, buffer, 0, fieldSize);
+
+            jsonString.Append(buffer[0..(int)(fieldSize)]);
```

Comment:
Casting fieldSize to int seems redundant as mentioned by intellisense.

Reply:
do you also need to accommodate for the out of range exception that this can throw?

Reply:
Changed a bit of the logic here and fixed.